### ISSUE TITLE
Broken error messages cause by 

### ISSUE THREAD

**Issue by bogdandrutu** [go-viper/mapstructure#49]
See https://github.com/open-telemetry/opentelemetry-collector/pull/11427, I had to do this to be able to upgrade to v2.2.1.

This is most likely caused by https://github.com/go-viper/mapstructure/commit/b0e65b30997c3f9e72b4b2d58b281377853fadbf#diff-7276b880a688d3af79f0a7b18e13df164968ac39766a475f3c91f9c445558e8fR803

/cc @yurishkuro

**Comment by sagikazarmark** [comment]
@yurishkuro looking at that change, I'm not quite sure why it's necessary. Can you clarify? Is it because of the nil value use case?

**Comment by yurishkuro** [comment]
yes, the original error message did not work under all conditions (it would panic)
